<!DOCTYPE html>
<html lang="en">

<head>

  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
  <meta name="description" content="">
  <meta name="author" content="">
  <title>CS 501 - Computational Biology</title>

  <!-- Bootstrap core CSS -->
  <link href="vendor/bootstrap/css/bootstrap.min.css" rel="stylesheet">

  <!-- Custom fonts for this template -->
  <link href="vendor/fontawesome-free/css/all.min.css" rel="stylesheet" type="text/css">
  <link href="https://fonts.googleapis.com/css?family=Montserrat:400,700" rel="stylesheet" type="text/css">
  <link href="https://fonts.googleapis.com/css?family=Lato:400,700,400italic,700italic" rel="stylesheet" type="text/css">

  <!-- Plugin CSS -->
  <link href="vendor/magnific-popup/magnific-popup.css" rel="stylesheet" type="text/css">

  <!-- Custom styles for this template -->
  <link href="css/freelancer.min.css" rel="stylesheet">


  <script src="https://cdn.anychart.com/releases/8.3.0/js/anychart-base.min.js"></script>
  <link ref="https://cdn.anychart.com/releases/8.3.0/fonts/css/anychart-font.min.css" />
  <script>
    anychart.onDocumentReady(function () {
      // create data set on our data
      var dataSet = anychart.data.set(getData());

      // map data for the first series, take x from the zero column and value from the first column of data set
      var seriesData_1 = dataSet.mapAs({ 'x': 0, 'value': 1 });

      // map data for the second series, take x from the zero column and value from the second column of data set
      var seriesData_2 = dataSet.mapAs({ 'x': 0, 'value': 2 });

      // create line chart
      var chart = anychart.line();

      // turn on chart animation
      chart.animation(true);

      // set chart padding
      chart.padding([10, 20, 5, 20]);

      // turn on the crosshair
      chart.crosshair()
        .enabled(true)
        .yLabel(false)
        .yStroke(null);

      // set tooltip mode to point
      chart.tooltip().positionMode('point');

      // set chart title text settings
      chart.title('Manhattan and Euclidean Similarity Between Iris-Versicolor pairs');

      // set yAxis title
      chart.yAxis().title('Similarity');
      chart.xAxis().labels().padding(5);

      // create first series with mapped data
      var series_1 = chart.line(seriesData_1);
      series_1.name('Euclidean Distance');
      series_1.hovered().markers()
        .enabled(true)
        .type('circle')
        .size(4);
      series_1.tooltip()
        .position('right')
        .anchor('left-center')
        .offsetX(5)
        .offsetY(5);
      series_1.stroke('red');

      // create second series with mapped data
      var series_2 = chart.line(seriesData_2);
      series_2.name('Manhattan Distance');
      series_2.hovered().markers()
        .enabled(true)
        .type('circle')
        .size(4);
      series_2.tooltip()
        .position('right')
        .anchor('left-center')
        .offsetX(5)
        .offsetY(5);


      // turn the legend on
      chart.legend()
        .enabled(true)
        .fontSize(13)
        .padding([0, 0, 10, 0]);

      // set container id for the chart
      chart.container('container');
      // initiate chart drawing
      chart.draw();
    });

    function getData() {
      return [
        ['a & b', 0.640312423743, 0.9],
        ['b & c', 0.648074069841, 1.0],
        ['c & d', 1.8574175621, 3.3],
        ['d & e', 1.28452325787, 2.3],
        ['e & f', 0.830662386292, 1.1],
        ['f & g', 0.860232526704, 1.6],
        ['g & h', 2.25610283454, 4.3],
        ['h & i', 2.21810730128, 3.8],
        ['i & j', 1.58113883008, 2.4],
        ['j & k', 0.921954445729, 1.7],
        ['k & l', 1.59687194227, 3.1],
        ['l & m', 0.969535971483, 1.6],
        ['m & n', 1.07238052948, 1.9],
        ['n & o', 1.2124355653, 1.7],
        ['o & p', 1.37840487521, 2.2],
        ['p & q', 1.11355287257, 1.4],
        ['q & r', 0.734846922835, 1.4],
        ['r & s', 0.905538513814, 1.8],
        ['s & t', 0.98488578018, 1.9],
        ['t & u', 1.37113092008, 2.6],
        ['u & v', 1.04403065089, 1.9],
        ['v & w', 0.989949493661, 1.6],
        ['w & x', 0.509901951359, 1.0],
        ['x & y', 0.519615242271, 0.9],
        ['y & z', 0.264575131106, 0.5],
      ]
    }
  </script>

</head>


<style>
  .contributors {
    color:#2c3e50
  }
  .contributors h4 {
    margin-top:20px;
  }
  a#link-to-dataset:hover {
    color: black;
}
  a#link-to-dataset {
    color: white;
    text-decoration: underline;
}
table a:link {
	color: #666;
	font-weight: bold;
	text-decoration:none;
}
table a:visited {
	color: #999999;
	font-weight:bold;
	text-decoration:none;
}
table a:active,
table a:hover {
	color: #bd5a35;
	text-decoration:underline;
}
table {
	font-family:Arial, Helvetica, sans-serif;
	color:#666;
	font-size:12px;
	text-shadow: 1px 1px 0px #fff;
	background:#eaebec;
	margin:20px;
	border:#ccc 1px solid;

	-moz-border-radius:3px;
	-webkit-border-radius:3px;
	border-radius:3px;

	-moz-box-shadow: 0 1px 2px #d1d1d1;
	-webkit-box-shadow: 0 1px 2px #d1d1d1;
	box-shadow: 0 1px 2px #d1d1d1;
}
table th {
	padding:21px 25px 22px 25px;
	border-top:1px solid #fafafa;
	border-bottom:1px solid #e0e0e0;

	background: #ededed;
	background: -webkit-gradient(linear, left top, left bottom, from(#ededed), to(#ebebeb));
	background: -moz-linear-gradient(top,  #ededed,  #ebebeb);
}
table th:first-child {
	text-align: left;
	padding-left:20px;
}
table tr:first-child th:first-child {
	-moz-border-radius-topleft:3px;
	-webkit-border-top-left-radius:3px;
	border-top-left-radius:3px;
}
table tr:first-child th:last-child {
	-moz-border-radius-topright:3px;
	-webkit-border-top-right-radius:3px;
	border-top-right-radius:3px;
}
table tr {
	text-align: center;
	padding-left:20px;
}
table td:first-child {
	text-align: left;
	padding-left:20px;
	border-left: 0;
}
table td {
	padding:18px;
	border-top: 1px solid #ffffff;
	border-bottom:1px solid #e0e0e0;
	border-left: 1px solid #e0e0e0;

	background: #fafafa;
	background: -webkit-gradient(linear, left top, left bottom, from(#fbfbfb), to(#fafafa));
	background: -moz-linear-gradient(top,  #fbfbfb,  #fafafa);
}
table tr.even td {
	background: #f6f6f6;
	background: -webkit-gradient(linear, left top, left bottom, from(#f8f8f8), to(#f6f6f6));
	background: -moz-linear-gradient(top,  #f8f8f8,  #f6f6f6);
}
table tr:last-child td {
	border-bottom:0;
}
table tr:last-child td:first-child {
	-moz-border-radius-bottomleft:3px;
	-webkit-border-bottom-left-radius:3px;
	border-bottom-left-radius:3px;
}
table tr:last-child td:last-child {
	-moz-border-radius-bottomright:3px;
	-webkit-border-bottom-right-radius:3px;
	border-bottom-right-radius:3px;
}
table tr:hover td {
	background: #f2f2f2;
	background: -webkit-gradient(linear, left top, left bottom, from(#f2f2f2), to(#f0f0f0));
	background: -moz-linear-gradient(top,  #f2f2f2,  #f0f0f0);
}

</style>

<body id="page-top">

  <!-- Navigation -->
  <nav class="navbar navbar-expand-lg bg-secondary fixed-top text-uppercase" id="mainNav">
    <div class="container">
      <a class="navbar-brand js-scroll-trigger" href="#page-top">CS 502 - Comp Biology</a>
      <button class="navbar-toggler navbar-toggler-right text-uppercase bg-primary text-white rounded" type="button"
        data-toggle="collapse" data-target="#navbarResponsive" aria-controls="navbarResponsive" aria-expanded="false"
        aria-label="Toggle navigation">
        Menu
        <i class="fas fa-bars"></i>
      </button>
      <div class="collapse navbar-collapse" id="navbarResponsive">
        <ul class="navbar-nav ml-auto">
          <li class="nav-item mx-0 mx-lg-1">
            <a class="nav-link py-3 px-0 px-lg-3 rounded js-scroll-trigger" href="#contributors">Contributors</a>
          </li>
          <li class="nav-item mx-0 mx-lg-1">
            <a class="nav-link py-3 px-0 px-lg-3 rounded js-scroll-trigger" href="#similarity">Similarity</a>
          </li>
          <li class="nav-item mx-0 mx-lg-1">
            <a class="nav-link py-3 px-0 px-lg-3 rounded js-scroll-trigger" href="#PaperReview">Review</a>
          </li>
          <li class="nav-item mx-0 mx-lg-1">
            <a class="nav-link py-3 px-0 px-lg-3 rounded js-scroll-trigger" href="#proposal">Proposal</a>
          </li>
          <li class="nav-item mx-0 mx-lg-1">
            <a class="nav-link py-3 px-0 px-lg-3 rounded js-scroll-trigger" href="#progress">Progress</a>
          </li>
          <!-- <li class="nav-item mx-0 mx-lg-1">
            <a class="nav-link py-3 px-0 px-lg-3 rounded js-scroll-trigger" href="#glossary">Glossary</a>
          </li> -->
        </ul>
      </div>
    </div>
  </nav>

  <!-- Header -->
  <header class="masthead bg-primary text-white text-center">
    <div class="container">
      <img class="img-fluid mb-5 d-block mx-auto" style="width: 500px;" src="img/bioinfomatics_venn_diagram.png" alt="">
      <h1 class="text-uppercase mb-0">Computational Biology</h1>
      <hr class="star-light">
      <h2 class="font-weight-light mb-0">Amna Irfan - Shivani Kamath Belman - Pratik Kshirsagar</h2>
    </div>
  </header>

  <!-- Portfolio Grid Section -->
  <section class="portfolio" id="contributors">
    <div class="container">
      <h2 class="text-center text-uppercase text-secondary mb-0">Contributors</h2>
      <hr class="star-dark mb-5">
      <div class="row">
        <div class="col-md-6 col-lg-4 contributors" style="text-align: justify;">
          <img class="rounded-circle" src="img/amna.jpg" alt="Amna Irfan" width="190" height="200">
          <h4>Amna Irfan</h4>
          <p>I am a first year Computer Science graduate student at UIC. Last year, I graduated with a bachelors degree
            in Computer Science and a minor in Linguistics. Since the past 3 years, I have been working as a full-stack
            developer at LeadFuze which is a marketing startup. In terms of my undergraduate research experience, I
            have worked on statistical topic models (mostly LDA) and dialogue systems. I am fluent in Java, Python,
            GoLang, C#. During the research analysis phase of my projects, I also had the chance to work in R-studio
            and SAS.
            Why am I taking this course? I am obsessed with animals specially cats and I wanted to learn a bit more on
            how computer science can be applied to the life sciences. Right now, I am set on doing my graduate research
            thesis on NLP, but I want to see if this course will change my mind.
          </p>
        </div>
        <div class="col-md-6 col-lg-4 contributors" style="text-align: justify;">
          <img class="rounded-circle" src="img/shivaniphoto.jpg" alt="Shivani Belman" width="190" height="200">
          <h4>Shivani Kamath Belman</h4>
          <p>Hello there :) I am a Master's student in the Bioengineering department at UIC with a specialization in
            Bioinformatics. I have a Bachelor's degree in Computer Science. </br> I always enjoyed biology so I chose
            to get into bioinformatics to apply my computer science skillset in a field I like. I have experience
            working with various research teams and medical data. </br>I chose this course because it allows me to
            delve into the computational aspects of biology.</p>
        </div>
        <div class="col-md-6 col-lg-4 contributors" style="text-align: justify;">
          <img class="rounded-circle" src="img/Pratik.jpg" alt="Pratik Kshirsagar" width="190" height="200">
          <h4>Pratik Kshirsagar</h4>
          <p>I'm a Master's student in Computer Science. I like to understand why things work the way they do. ü§î</p>
          <p></p>I love science for it can show how intelligent ü§ì (or dumb ü§∑) humans can be ;)</p>
          <p>Curiosity killed the cat but satisfaction brought it back. üòã => Let nothing hold you back from learning!</p>
          <p>I took this course out of my geniune love for biology since an early age. I hope to acquire new knowledge
            and have some fun along the way!</p>
        </div>
      </div>
    </div>
  </section>

  <section class="bg-primary text-white mb-0" id="similarity">
    <div class="container">
      <h2 class="text-center text-uppercase text-white">Similarity</h2>
      <hr class="star-light mb-5">
      <div class="row">
        <div class="col-lg-12 ml-auto">
          <h4 class="text-center">Problem Statement</h4>
          <p class="lead">Find a biological database of your choice and download 2 closely related entries from it.
            How did you determine that the entries are closely related? What is the ‚Äúdistance‚Äù between the entries?
            How did you define and compute it?</p>
          <h4 class="text-center">Dataset</h4>
          <p class="lead">In our very first task related to the project, we have selected an exceedingly simple domain.
            The data we have selected
            is the 'Iris Data Set'. This dataset has information on various physical traits of flowers from different
            species belonging to the Iris genus.</p>
          <p class="lead">The dataset can be found <a id="link-to-dataset" href="https://archive.ics.uci.edu/ml/datasets/Iris"
              target="_blank">here.</a></p>
          <h4 class="text-center">Dataset Description</h4>
          <p class="lead">The dataset we have picked contains information about samples belonging to the Iris genus of
            flowers.
            Based on their attributes or traits, we can predict what species they belong to, hence being an ideal
            dataset for learning
            classification.</p>
          <p class="lead">There are 150 instances of different flowers. For each instance, there exist 4 attributes and
            a class attribute which
            tells us which species the instance actually belongs to.
            Each attribute is a numerical value that quantifies the particular trait and the class label is a
            categorical value.</p>

          <h5>The 4 attributes are:</h5>
          <ul class="lead">
            <li>Sepal width in cm
            <li>Sepal length in cm
            <li>Petal width in cm
            <li>Petal length in cm
          </ul>

          <h5>The 3 class attributes are:</h5>
          <ul class="lead">
            <li>Iris Setosa
            <li>Iris Virginica
            <li>Iris Versicolour
          </ul>

          <p class="lead text-center">Below there is an example of how the dataset looked for the first few instances.</p>
          <img class="img-fluid mb-5 d-block mx-auto" style="width: 800px;" src="./img/Sample - Original.png">

          <h4 class="text-center">Preliminary Analysis and Preprocessing</h4>
          <p class=" text-justify lead">Our initial analysis allowed us to conclude that there were no missing values.
            This makes our data as accuracte and
            reliable as possible.
            We also found out that there were no duplicate values. This is great news because this avoids any confusion
            due to
            unusual similarity scores and helps avoid unnecessary computations.
            Since every data instance was unique, finding similarity between 2 entries was extremely convenient and
            would give
            reliable values.</p>

          <p class=" text-justify lead">As it can be seen from the image of sample data above, all of the attributes
            are useful in calculating similarity.
            (Hooray!)
            Every instance, ie, flower, has a unique sepal width, sepal length, petal width and petal length. All of
            these
            properties can be used to calculate similarity.
            The class label too is instrumental in finding out whether 2 instances are similar or not, but in order to
            keep a
            uniform measure of similarity, we decided to exclude it from our initial computations.</p>

          <p class=" text-justify lead">We used a python script to import our dataset and prepare it (exclude the class
            label) for computing similarity, using pandas.</p>

          <img class="img-fluid mb-5 d-block mx-auto" style="width: 800px;" src="./img/Sample - Distance.png">

          <h4 class="text-center"></h4>
          <p class=" text-justify lead">After preliminary analysis, it was easy to conclude that we needed to find some
            sort of distance measure. We tried out
            Manhattan distance as well as Euclidean distance between the attribute values of
            different instances.</p>
          <p class=" text-justify lead">Our data instances have 4 attribute values each. Each of those values being
            numerical in nature. So, the best way to
            find similarity between 2 entries was
            to treat both of the instances as vectors [1-D arrays] and find the Manhattan as well as Euclidean
            distances between
            the 2 instances.</p>

          <p class="text-center lead">Manhattan distance = Sum(|Val ij - Val ik|)</p>
          <p class="text-center lead">Euclidean distance = Sq.
            root(Sum(|Val ij - Val ik|^2)).</p>
          <p class="text-center lead">where Val ij = value of attribute i for data
            instance j and
            Val ik = value of attribute i for data instance k.</p>

          <p class=" text-justify lead">Both of these turned out equally effective distance measures but it seemed more
            convenient to consider Euclidean as our
            primary distance measure purely
            due to convention. As we all know, the L2 Norm is the most popular distance measure.</p>

          <p class=" text-center lead">We achieved these aforementioned computations through simple dataframe
            operations in Python.</p>
          <img class="img-fluid mb-5 d-block mx-auto" style="width: 800px;" src="./img/Processed.png">
          <img class="img-fluid mb-5 d-block mx-auto" style="width: 800px;" src="./img/code1.png">
          <img class="img-fluid mb-5 d-block mx-auto" style="width: 800px;" src="./img/code2.png">

          <p class=" text-justify lead">So as we can see from the above example, the Manhattan distance between the two
            entries [73] and [98] is 1.997 and 3.099 respectively.</p>

          <p class=" text-justify lead">We used the 'distance' package from SciPy to calculate both the Euclidean and
            Manhattan distances. The cityblock and euclidean functions used in the example above, are from the said
            package.</p>
          <br />
          <h4 class="text-center">Correlation Between Euclidean and Manhattan Distance</h4>

          <p class=" text-justify lead">The following table represents the first 26 records from the flower class
            Iris-Versicolor.</p>

          <table class="mx-auto" cellspacing='0'>
            <!-- cellspacing='0' is important, must stay -->

            <!-- Table Header -->
            <thead>
              <tr>
                <th>ID</th>
                <th>Sepal Length</th>
                <th>Sepal Width</th>
                <th>Petal Length</th>
                <th>Petal Width</th>
                <th>Class</th>
              </tr>
            </thead>
            <!-- Table Header -->

            <!-- Table Body -->
            <tbody>
              <tr class='even'>
                <td>a</td>
                <td>7.0</td>
                <td>3.2</td>
                <td>4.7</td>
                <td>1.4</td>
                <td>Iris-versicolor</td>
              </tr>
              <tr class=''>
                <td>b</td>
                <td>6.4</td>
                <td>3.2</td>
                <td>4.5</td>
                <td>1.5</td>
                <td>Iris-versicolor</td>
              </tr>
              <tr class='even'>
                <td>c</td>
                <td>6.9</td>
                <td>3.1</td>
                <td>4.9</td>
                <td>1.5</td>
                <td>Iris-versicolor</td>
              </tr>
              <tr class=''>
                <td>d</td>
                <td>5.5</td>
                <td>2.3</td>
                <td>4.0</td>
                <td>1.3</td>
                <td>Iris-versicolor</td>
              </tr>
              <tr class='even'>
                <td>e</td>
                <td>6.5</td>
                <td>2.8</td>
                <td>4.6</td>
                <td>1.5</td>
                <td>Iris-versicolor</td>
              </tr>
              <tr class=''>
                <td>f</td>
                <td>5.7</td>
                <td>2.8</td>
                <td>4.5</td>
                <td>1.3</td>
                <td>Iris-versicolor</td>
              </tr>
              <tr class='even'>
                <td>g</td>
                <td>6.3</td>
                <td>3.3</td>
                <td>4.7</td>
                <td>1.6</td>
                <td>Iris-versicolor</td>
              </tr>
              <tr class=''>
                <td>h</td>
                <td>4.9</td>
                <td>2.4</td>
                <td>3.3</td>
                <td>1.0</td>
                <td>Iris-versicolor</td>
              </tr>
              <tr class='even'>
                <td>i</td>
                <td>6.6</td>
                <td>2.9</td>
                <td>4.6</td>
                <td>1.3</td>
                <td>Iris-versicolor</td>
              </tr>
              <tr class=''>
                <td>j</td>
                <td>5.2</td>
                <td>2.7</td>
                <td>3.9</td>
                <td>1.4</td>
                <td>Iris-versicolor</td>
              </tr>
              <tr class='even'>
                <td>k</td>
                <td>5.0</td>
                <td>2.0</td>
                <td>3.5</td>
                <td>1.0</td>
                <td>Iris-versicolor</td>
              </tr>
              <tr class=''>
                <td>l</td>
                <td>5.9</td>
                <td>3.0</td>
                <td>4.2</td>
                <td>1.5</td>
                <td>Iris-versicolor</td>
              </tr>
              <tr class='even'>
                <td>m</td>
                <td>6.0</td>
                <td>2.2</td>
                <td>4.0</td>
                <td>1.0</td>
                <td>Iris-versicolor</td>
              </tr>
              <tr class=''>
                <td>n</td>
                <td>6.1</td>
                <td>2.9</td>
                <td>4.7</td>
                <td>1.4</td>
                <td>Iris-versicolor</td>
              </tr>
              <tr class='even'>
                <td>o</td>
                <td>5.6</td>
                <td>2.9</td>
                <td>3.6</td>
                <td>1.3</td>
                <td>Iris-versicolor</td>
              </tr>
              <tr class=''>
                <td>p</td>
                <td>6.7</td>
                <td>3.1</td>
                <td>4.4</td>
                <td>1.4</td>
                <td>Iris-versicolor</td>
              </tr>
              <tr class='even'>
                <td>q</td>
                <td>5.6</td>
                <td>3.0</td>
                <td>4.5</td>
                <td>1.5</td>
                <td>Iris-versicolor</td>
              </tr>
              <tr class=''>
                <td>r</td>
                <td>5.8</td>
                <td>2.7</td>
                <td>4.1</td>
                <td>1.0</td>
                <td>Iris-versicolor</td>
              </tr>
              <tr class='even'>
                <td>s</td>
                <td>6.2</td>
                <td>2.2</td>
                <td>4.5</td>
                <td>1.5</td>
                <td>Iris-versicolor</td>
              </tr>
              <tr class=''>
                <td>t</td>
                <td>5.6</td>
                <td>2.5</td>
                <td>3.9</td>
                <td>1.1</td>
                <td>Iris-versicolor</td>
              </tr>
              <tr class='even'>
                <td>u</td>
                <td>5.9</td>
                <td>3.2</td>
                <td>4.8</td>
                <td>1.8</td>
                <td>Iris-versicolor</td>
              </tr>
              <tr class=''>
                <td>v</td>
                <td>6.1</td>
                <td>2.8</td>
                <td>4.0</td>
                <td>1.3</td>
                <td>Iris-versicolor</td>
              </tr>
              <tr class='even'>
                <td>w</td>
                <td>6.3</td>
                <td>2.5</td>
                <td>4.9</td>
                <td>1.5</td>
                <td>Iris-versicolor</td>
              </tr>
              <tr class=''>
                <td>x</td>
                <td>6.1</td>
                <td>2.8</td>
                <td>4.7</td>
                <td>1.2</td>
                <td>Iris-versicolor</td>
              </tr>
              <tr class='even'>
                <td>y</td>
                <td>6.4</td>
                <td>2.9</td>
                <td>4.3</td>
                <td>1.3</td>
                <td>Iris-versicolor</td>
              </tr>

              <tr class=''>
                <td>z</td>
                <td>6.6</td>
                <td>3.0</td>
                <td>4.4</td>
                <td>1.4</td>
                <td>Iris-versicolor</td>
              </tr>

            </tbody>
            <!-- Table Body -->

          </table>

          <p class=" text-justify lead">Plotting Manhattan and Euclidean similarity distance between nth - 1 and nth
            record where n = 26.</p>

          <div id="container" style="height:400px"></div>
        </div>
        <br />
        <p class="text-justify lead" style="margin-top:50px;">Looking at the line graph above, there seems to be a
          correlation between the Manhattan and Euclidean similarity measures. We will now use a scatter plot to plot
          Manhattan distance as a function of Euclidean distance and find if they correlate. If they do, we will also
          try to find a regression line that best describes their relation. Note that N = 51</p>

        <div id="scatter_plot" class="mx-auto" style="width:100%;height:500px;margin-top:50px;"></div>
        <script src="https://cdn.anychart.com/samples/scatter-charts/combination-of-line-and-marker-charts/data.js"></script>
        <script src="https://cdn.anychart.com/releases/8.3.0/js/anychart-base.min.js"></script>
        <script src="https://cdn.anychart.com/releases/8.3.0/js/anychart-ui.min.js"></script>
        <script src="https://cdn.anychart.com/releases/8.3.0/js/anychart-exports.min.js"></script>
        <script type="text/javascript">anychart.onDocumentReady(function () {
            var chart = anychart.scatter();

            // turn on chart animation
            chart.animation(true);

            // set chart title
            chart.title('Correlation Between Manhattan and Euclidean Similarity Distance');

            chart.xScale()
              .minimum(0)
              .maximum(3)
              .ticks({ interval: 0.05 });
            chart.yScale()
              .minimum(0)
              .maximum(5)
              .ticks({ interval: 0.05 });

            chart.yAxis().title('Manhattan Similarity');
            chart.xAxis()
              .title('Euclidean Similarity')
              .drawFirstLabel(false)
              .drawLastLabel(false);

            chart.interactivity()
              .hoverMode('by-spot')
              .spotRadius(30);

            chart.tooltip().displayMode('union');

            // The data used in this sample can be obtained from the CDN
            // https://cdn.anychart.com/samples/scatter-charts/combination-of-line-and-marker-charts/data.js
            var marker = chart.marker([
              [0.640312423743, 0.9],
              [0.648074069841, 1.0],
              [1.8574175621, 3.3],
              [1.28452325787, 2.3],
              [0.830662386292, 1.1],
              [0.860232526704, 1.6],
              [2.25610283454, 4.3],
              [2.21810730128, 3.8],
              [1.58113883008, 2.4],
              [0.921954445729, 1.7],
              [1.59687194227, 3.1],
              [0.969535971483, 1.6],
              [1.07238052948, 1.9],
              [1.2124355653, 1.7],
              [1.37840487521, 2.2],
              [1.11355287257, 1.4],
              [0.734846922835, 1.4],
              [0.905538513814, 1.8],
              [0.98488578018, 1.9],
              [1.37113092008, 2.6],
              [1.04403065089, 1.9],
              [0.989949493661, 1.6],
              [0.509901951359, 1.0],
              [0.519615242271, 0.9],
              [0.264575131106, 0.5],
              [0.489897948557, 0.8],
              [0.424264068712, 0.8],
              [0.888819441732, 1.5],
              [1.19582607431, 2.1],
              [0.424264068712, 0.8],
              [0.141421356237, 0.2],
              [0.509901951359, 1.0],
              [1.28062484749, 1.8],
              [0.905538513814, 1.6],
              [0.728010988928, 1.1],
              [0.793725393319, 1.3],
              [0.964365076099, 1.7],
              [1.03440804328, 1.7],
              [0.519615242271, 0.7],
              [0.424264068712, 0.6],
              [0.774596669241, 1.4],
              [0.80622577483, 1.5],
              [1.12249721603, 2.0],
              [1.19163752878, 2.2],
              [0.331662479036, 0.5],
              [0.141421356237, 0.2],
              [0.509901951359, 0.6],
              [1.76068168617, 3.0],
              [1.30384048104, 2.2],
            ]);
            marker.type('triangle-up')
              .size(4);
            marker.hovered()
              .size(7)
              .fill('gold')
              .stroke(anychart.color.darken('gold'));
            marker.tooltip()
              .hAlign('start')
              .format(function () {
                return 'Manhattan: ' + this.value + '\nEuclidean: ' + this.x;
              });

            // The data used in this sample can be obtained from the CDN
            // https://cdn.anychart.com/samples/scatter-charts/combination-of-line-and-marker-charts/data.js
            chart.line([
              [0, -0.074],
              [1, 1.71],
              [3, 5.278]
            ]);
            //y = ‚àí0.074 + 1.784‚ãÖx
            // set container id for the chart
            chart.container('scatter_plot');
            // initiate chart drawing
            chart.draw();
          });</script>
      </div>
      <br />
      <p class=" text-justify lead">
        <b>Result</b>
        <p>R Calculation</p>
        <p>r = 19.87 / ‚àö((11.138)(37.087)) = 0.9776</p>
        <p><u>The value of R is 0.9776. Therefore, in the domain of Iris, there is a strong positive correlation
            between the Manhattan and Euclidean similarity distance.</u></p>
        <b>Regression Line: y = ‚àí0.074 + 1.784‚ãÖx</b>
      </p>

    </div>
  </section>
  <section class="portfolio" id="PaperReview">
    <div class="container">
      <h2 class="text-center text-uppercase text-secondary mb-0">Paper Review</h2>
      <hr class="star-dark mb-5">
      <div class="row">
        <div class="col-lg-12 ml-auto">
          <h4 class="text-center">Problem Statement</h4>
          <p class="text-justify lead">We were instructed to review a paper from the RECOMB conference. So, we chose a
            paper from RECOMB 2013 titled ‚ÄúIPED: Inheritance Path-based Pedigree Reconstruction Algorithm
            Using Genotype Data‚Äù. The paper can be found <a href="./img/pedigree.pdf">here.</a> </p>
          <h4 class="text-center">Paper Summary</h4>
          <p class="text-justify lead">The paper focuses on finding an efficient way of pedigree reconstruction, ie,
            inference of family trees, which is a
            fundamental problem in genetics. It talks about the drawbacks of existing algorithms with respect to space
            and time
            complexity. Existing algorithms perform poorly when the number of generations is greater than a certain
            threshold, 4
            according to the paper, and also perform poorly for inbred populations. The paper puts forward an algorithm
            that
            handles cases of both inbred and outbred populations as well as really large number of generations,
            efficiently.
          </p>
          <p class="text-justify lead">The paper first introduced us to the concept of pedigree graphs and all the
            necessary nomenclature associated with
            them. It then talks about evaluating the relationship of pairs of individuals for both kind of pairs;
            extant and
            ancestral. The IBD(identity-by-descent) measure is central to this evaluation.
          </p>
          <img class="img-fluid mb-5 d-block mx-auto" style="width: 800px;" src="./img/relation.png">
          <p class="text-justify lead">
            For outbred populations, M = 2(g - 1) : g is the generation, but for inbred populations, M is calculated by
            an algorithm.

            For chromosomes in (i, j), they calculate the average V<sub>i, j</sub> over all combinations of i, j.
          </p>
          <p class="text-justify lead"> To determine relationship of an ancestral pair (k, l), a similar strategy is
            used but the weighted average of
            relationship score of all possible pairs of chromosomes in both the ancestral pair is calculated.
          </p>
          <img class="img-fluid mb-5 d-block mx-auto" style="width: 500px;" src="./img/ancestor.png">
          <p class="text-justify lead">
            Also, to find E (IBD) and var (IBD), one needs to find the number of meoises (M). The authors have proposed
            an algorithm that deals with inheritance
            paths and their lengths from a shared ancestor. They use Inheritance Path Pairs(IPPs) to represent the
            length of the
            path and the number of such paths between an ancestor and an extant. They use a hash table for storing
            these IPP
            tuples, which speed up their time complexity. To compute these IPPs they use a Dynamic Programming
            algorithm that uses
            the IPPs available for current generation to calculate IPPs for next generation. The IPPs for a child are
            just a merge
            of IPPs of their parent. This algorithm has time complexity linear in terms of pedigree height.
          </p>
          <p class="text-justify lead">
            Given the IPPs for two individuals, the number of Meioses (M) can be calculated which in turn can be used
            to calculate
            the expected value and variance of IBDs.

            Once the relationship scores have been calculated, the siblings can be assigned to the same parents with
            the help of a
            Max-Clique algorithm that finds a maximum clique of siblings.

            Only if the distance between a pair of extant individuals in the original and the reconstructed pedigree
            are the same,
            the reconstruction is considered accurate.

            Results from the paper suggest that IPED was better at pedigree chart generation and also faster compared
            to the
            traditional COP in Outbreeding cases, it also does not rely on any threshold value. In case of the
            inbreeding
            instances, IPED performs far better than CIP or COP algorithms as the number of generations increases. In
            conclusion,
            the authors state that IPED is an very efficient algorithm for pedigree reconstruction, it is independent
            of threshold
            values and sampling. They specifically focus on how well the algorithm performs for inbreeding cases.
          </p>
          <h4 class="text-center">Discussion Summary</h4>
          <p class="text-justify lead">
            We think that the problem was interesting and fun. We thought that the IPED did a great job when it was
            tested against
            the Inbred cases, but not having considered ‚Äòhalf-siblings‚Äô makes the problem less complex and incomplete.
            Also the authors stated that the algorithm is not optimal, and going for the approximation algorithm makes
            the
            algorithm vulnerable to breaking for specific cases, such as the presence of common inheritance path pairs
            amongst
            individuals. The approximation of the mean and variance of IBD length from average number of meiosis
            renders it
            nonoptimal.
          </p>
          <p class="text-justify lead">
            We think that the approach is commendable but we do not have in-depth knowledge about the cases in question
            yet so we
            don‚Äôt have a case to present for a better approach. We were impressed with the drastic improvement in
            accuracy that
            IPED brought about in reconstructing pedigree graphs for inbred populations.
          </p>
        </div>
      </div>
    </div>
  </section>

  <section class="bg-primary text-white mb-0" id="proposal">
    <div class="container">
      <h2 class="text-center text-uppercase text-white">Proposal</h2>
      <hr class="star-light mb-5">
      <div class="row">
        <div class="col-lg-12 ml-auto">
          <h4 class="text-center">Introduction</h4>
          <br />
          <p class="text-justify lead"><u>Clustered regularly interspaced short palindromic repeats known as CRISPR</u>
            was
            first found in bacterial/prokaryotic cells. It is a built-in natural immune system that the bacterial cell
            uses to fight against viral invasions within its body.
          </p>
          <p class="text-justify lead">In 1980s, scientists started noticing a strange pattern in some bacterial
            genomes. They found short
            palindromic repeating sequences within the genome that were 28-37 base pairs long each and were separated
            by other short sequences (32-38 base pairs) which were later named ‚ÄúSpacer Sequences/DNA‚Äù. These spaces
            turned out to be copies of the DNA from viruses that previously attacked the bacteria or its ancestors.
            These repeats along with the spaces together are known as the ‚ÄúCRISPR locus‚Äù.</p>
          <p class="text-justify lead">During a phage attack, the virus modifies the bacterial genome by injecting its
            own DNA into the bacterial
            cell. In defense, the bacteria responds by activating the CRISPR CAS9 system which is a natural defense
            mechanism within its cell. There are 3 steps to this immune system.</p>
          <ul class="text-justify lead">
            <li><u>Spacer Acquisition:</u> This step only takes place when the bacteria encounters an unknown virus for
              the first time. The bacterial cell cuts a piece from the viral genome via CAS9 Enzymes (usually helicases
              and nucleases) and inserts it into the spacer DNA slots. This is similar to taking a mugshot of the
              bacteriophage.</li><br>
            <li><u>CrRna Processing:</u> If the bacteriophage genome is recognized by the bacterial cell, it will
              transcribe one of the strands from the CRISPR Locus to a crRNA which will serve as the ‚Äúguided RNA‚Äù in
              the next step.
            </li><br>
            <li><u>Interference:</u> The crRna is then integrated with the CAS9 protein. CAS9 finds the complementary
              DNA strand to the crRna (aka a single guided RNA or sgRna) and cuts/modifies that DNA sequence in the
              bacterial
              cell through homologous recombination or non-homologous end joining. This results in the bacterial cells
              removing the foreign viral genes from their genome.</li><br>
          </ul>
          <p class="text-justify lead">
            Since proteins are mostly universal, this discovery has opened up the possibility of editing/mutating the
            genome of any organism through injecting the CAS9 protein along with a guided RNA (CRISPR) into the target
            cells.
          </p>
          <h4 class="text-center">Problem</h4>
          <br />
          <p class="text-justify lead">Though CRISPR has been one of the greatest breakthroughs in technology, it comes
            with many down sides. It is one of the easiest gene engineering techniques discovered, but scientists have
            also discovered that it causes many mutations that are not supposed to occur. Changes in unintended target
            areas are often referred to as off-target effects.[4]
            One of the main reasons for off-target mutations is that CRISPR-Cas9 cleaves DNA sequences that are similar
            to the target sequences.[3] In order to reduce these problems we could try to maximize cleavage efficiency
            and minimize off-target effects.
            Our survey is going to consider various research papers that have been published to address the same. </p>

          <h4 class="text-center">Strategy</h4>
          <br />

          <p class="text-justify lead">Our aim is to learn in detail about CRISPR , how it works and survey the many
            machine learning methods
            that have been applied on making CRISPR cleavage efficient and predicting off-target effects of CRISPR.
            We plan to implement our survey gradually. First, we will focus on learning in depth about CRISPR-CAS9‚Äôs
            working mechanism, and how it is used in genetic engineering.
            Then we will focus on the machine learning methods that have been implemented to make CRISPR more
            efficient and accurate. We do not plan to implement the techniques due to lack of data and the lack of
            technology which is required to test the accuracy of methods even if they are implemented. We will put up
            our results on our website and also present our findings in class.</p>

          <h4 class="text-center">References</h4>
          <br />
          <ol style="padding: 0;">
            <li class="text-justify lead"> Abadi S, Yan WX, Amar D & Mayrose I (2017) A machine learning approach for
              predicting CRISPR-Cas9 cleavage efficiencies and patterns underlying its mechanism of action. PLoS Comput
              Biol 13(10): e1005807.</li>
            <li class="text-justify lead">Listgarten J,Weinstein M, Kleinstiver B P, Sousa A A,Joung J. K,
              Crawford J, Gao K, Hoang L, Elibol M, Doench J G & Fusi N. (2018) Prediction of
              off-target activities for the end-to-end design of CRISPR guide RNAs, Nature biomedical Engineering
              2:38-47.
            </li>
            <li class="text-justify lead">Doench J G, Fusi N, Sullender M, Hegde M, Vaimberg E W, Donovan K F,
              Smith I , Tothova Z, Wilen C, Orchard R, Virgin H W, Listgarten J & Root D E.(2016),
              Optimized sgRNA design to maximize activity and minimize off-target effects of CRISPR-Cas9, Nat
              Biotechnol,34(2):184-191.
            </li>
            <li class="text-justify lead"> Otieno MO (2015) CRISPR-Cas9 Human Genome Editing: Challenges, Ethical
              Concerns and Implications. J Clin Res Bioeth 6:253. doi: 10.4172/2155-9627.1000253

            </li>

          </ol>
          <br />
        </div>

      </div>
    </div>
  </section>

  <section class="portfolio" id="progress">
    <div class="container">
      <h2 class="text-center text-uppercase text-secondary mb-0">Progress</h2>
      <hr class="star-dark mb-5">
      <div class="row">
        <div class="col-lg-12 ml-auto">
          <h3><u>Summary 1:</u> RNA guided editing of bacterial genomes using CRISPR-Cas Systems</h4>
            <br />
            <p>
              This paper talks about how CRISPR systems with dual-RNA:Cas9 can be used for editing bacterial genomes.
              The
              authors argument mainly revolves about the fact that we can use this CRISPR system for genome editing
              instead of cleaving the genome at specific points.
            </p>
            <p>
              The authors reprogram the specificity by changing the sequence of short CRISPR RNA(crRNA) to make single
              and multi nucleotide changes carried on editing templates. The authors have worked with S. pneumoniae and
              E. coli as part of their analyses. It is shown that mutations can be introduced through the
              transformation
              of a template DNA fragment that will recombine in the genome and eliminate recognition of the target by
              the
              endonuclease (Cas9). Also it is observed that including several different crRNAs will let us introduce
              multiple mutations simultaneously.
            </p>
            <p>
              To introduce specific changes in the genome, one must use an editing template carrying mutations that
              avert
              cleavage by the endonuclease. This is easy to achieve when performing gene insertion. But, when gene
              fusion
              is desired or generation of single nucleotide mutations, this is possible only by introducing mutations
              in
              the editing template that alter either the PAM or protospacer sequences.
            </p>
            <p>
              While mutating the PAMs, there is only one criteria to be followed, we avoid mutating NGG to NAG or
              NNGGNN.
              All other random mutations in the PAMs are functional. Seed sequence (8-12bp adjacent to the PAMs) can
              prevent cleavage but there are restrictions with respect to nucleotide changing at each position and
              these
              restrictions vary with different spacers. Hence, mutations in the PAMs, if possible, should be the
              preferred strategy.
            </p>
            <p>
              The specificity and versatility of editing using the CRISPR-Cas system rely on several unique properties
              of
              the Cas9 endonuclease: (i) its target specificity can be programmed with a small RNA, without the need
              for
              enzyme engineering, (ii) target specificity is very high, determined by a 20-bp RNA-DNA interaction with
              low probability of nontarget recognition, (iii) almost any sequence can be targeted, the only requirement
              being the presence of an adjacent NGG sequence, (iv) almost any mutation in the NGG sequence, as well as
              mutations in the seed sequence of the protospacer, eliminates targeting.

            </p>
        </div>
      </div>
      <br />
      <div class="row">
        <div class="col-lg-12 ml-auto">
          <h3><u>Summary 2:</u> CCtop, An Intuitive, Flexible and Reliable CRISPR/Cas9 Target
            Prediction Tool</h4>
            <br />
            <p>
              The Crispr/Cas9 system technology is used in gene editing making it possible for knock-out, knock-in of
              genes, gene inactivation, <sup>&#8902;</sup> non-homologous end-joining as well as <sup>&#8902;</sup>homology
              directed repair. Though Crispr/Cas9 provides us with so many ways for gene editing, there are also a few
              drawbacks, one of them being the fact that there could be <sup>&#8902;</sup>off-target slicing of the
              gene.
              The authors interest is to develop an interactive tool, that is easy to use and can be used to evaluate
              and
              identify all candidate sgRNA target sites along with their off-target quality.
            </p>
            <p>
              To achieve this, the authors set up a web tool called Crispr/Cas9 target online prediction (CCtop) built
              with HTML and CGI scripts, using python for all the processing.
            </p>
            <p>
              They used the bowtie short read aligner to search for off target sites. It takes in index and few
              parameters along with it outputting the sequences that align with the input sequence. Then these
              alignments
              are scored to indicate the likelihood of the stability in the <sup>&#8902; </sup>sgRNA/DNA <sup>&#8902;
              </sup>heteroduplex. They also found that the likelihood decreases if the mismatch is closer to the PAM
              sequence.
            </p>
            <p>
              The formula used to score the off-target sequences is Score<sub>off-target</sub> = &#8721;<sub>mismatch</sub>
              1.2<sup>pos</sup>.
            </p>
            <p>
              Where pos is the position of each mismatch read from 5&#8217; end.
            </p>
            <p>
              After computing scores, each off target site is aligned with genes that its closest to. For this process,
              exon coordinate files of each organism are contained in the bx library in python which is used in this
              project. Basically, the off-target sites are checked to see if they contain any genes. Only exons closer
              than 100kb to target sites are assigned as the gene, else it is set to N/A. if they overlap, then the
              distance is set to 0. After assignment of genes, an sgRNA target site is scored. The sgRNA target sites
              are
              ranked based on how many numbers of predicted off-target sites are present and how they affect the
              off-target genes. The ranking is done using a single score which is calculated using the formula
            </p>
            <p>
              Score = &#8721;<sub>off-target</sub> = [log10(dist)+scoreoff-target]/ [total_off_targets] &#8722;
              total_off_targets.
            </p>
            <p>
              Where dist is the distance of each off-target site to corresponding closet exon.
              Off target sites with no associated exons are not considered for this score.
            </p>
            <p>
              Conclusion: The web tool developed was tested for gene inactivation and non-homology and homologous gene
              repair. The web tool is also constantly improved based on available resources. CCtop was designed to
              evaluate and present the users with possible sgRNA target sites, and the tool does what it was promised
              to
              do. It is also user friendly for first time amateur users at the same time it provides enough flexibility
              for experts to use it as well.
            </p>
            <p>
              Overall, the tool has been well received in the scientific community.
            </p>
        </div>
      </div>
      <br />
      <div class="row">
        <div class="col-lg-12 ml-auto">
          <h3><u>Summary 3:</u> CFD Score, Optimized sgRNA design to maximize activity and minimize off-target effects
            of CRISPR-Cas9
            </h4>
            <br />
            <p>
              CRISPR Cas9 is an adaptive autoimmune system present in bacterias to fight against phage attacks. This
              self-defense mechanism within the bacterial cell uses a Singular Guided RNA (sgRna) along with the Cas9
              endonuclease enzyme to eliminate bacteriophage/virus DNA within the cell. The Cas9 nuclease cleaves DNA
              that is complementary to the sgRNA. The concept of using an sgRNA with the Cas9 protein to remove DNA
              gave rise to the CRISPR Cas9 human-controlled gene-editing technique.
            </p>
            <p>
              Compared to other gene-editing techniques like Zinc finger nuclease and Transcription activator-like
              effector nucleases, CRISPR Cas9 is a more modern and simpler technique providing more versatility in
              genome engineering.
              However, given the age of the technique, CRISPR Cas9 has not been studied as extensively as older
              techniques. <br />
              Scientists found that CRISPR can cleave unwanted DNA sequences that are not an exact match to the input
              sgRNA. These mismatches known as off-targets can cause serious genetic damage.
            </p>
            <p>
              <u>Experiment Purpose & Design</u><br />
              <span>
                This paper investigates a number of techniques to design an optimal sgRNA to minimize off-targets and
                maximize activity. In this summary, we will discuss the sequence-related and point-wise mutation
                analysis the authors made while examining CRISPR Cas9 activity.<br />
                They conducted the CRISPR Cas9 experiment on mammalian cells using a library targeting the coding
                sequence of human CD33 with all possible sgRNAs, regardless of PAM.
              </span>
            </p>
            <p>
              <u>Results</u><br />
              They categorized sgRNA mutations in 3 ways.
              <ol>
                <li><b>Nucleotide Deletion: </b>This seldom led to activity. Out of 76 nucleotide and position
                  combinations, 14 retained high levels of cleavage activity from which 13 occured at positions 2, 3,
                  19 and 20. At positions 19 and 20, the deletion of cytosine in the RNA was seldom tolerated. </li>
                <li><b>Nucleotide Insertion: </b>Similar to deletion, mutation by nucleotide insertion rarely led to
                  activity. Most of the activity happened when nucleotides were inserted at position 2 and 3.</li>
                <li><b>Nucleotide Mismatch: </b>
                  <ul>
                    <li>89% of Guanine in RNA and Thymine in DNA mismatches were active across all positions.</li>
                    <li>37% of Cytosine in RNA and Cytosine in DNA mismatches were active across all positions.</li>
                  </ul>
                  Some single-base mismatches were position dependant.
                  <ul>
                    <li>At position 16, no Purine - Purine mismatches were tolerated but Purine - Thymine mismatches
                      were highly active.
                    </li>
                    <li>At position 19, 41% of Purine - Purine and 48% of Purine - Thymine mismatches were active.</li>
                  </ul>
                </li>
              </ol>
              They also found that sgRNAs with an NGG PAM were the most active.
            </p>
            <p>
              <u>Cutting Frequency Determination Scoring (CFD)</u><br/>
              <p>
                  The authors created a 2-dimensional matrix in which each cell represents the percent activity of a nucleotide base pair and mutation position. 
                  <br/><br/>The paper‚Äôs definition:<br/>
                  <blockquote>Cutting Frequency Determination (CFD) score is calculated by using the percent activity values from the matrix.  For example, if the interaction between the sgRNA and DNA has a single rG:dA mismatch in position 6, then that interaction receives a score of 0.67. If there are two or more mismatches, then individual mismatch values are multiplied together. For example, an rG:dA mismatch at position 7 coupled with an rC:dT mismatch at position 10 receives a CFD score of 0.57 √ó 0.87 = 0.50.</blockquote>
              </p>
            </p>
        </div>
      </div>
    </div>
  </section>

  <section class="bg-primary text-white mb-0" id="glossary">
    <div class="container">
      <h2 class="text-center text-uppercase text-white">Glossary</h2>
      <hr class="star-light mb-5">
      <div class="row">
        <div class="col-lg-12 ml-auto">
          <p class="text-justify lead">
            <u>Mutagenesis:</u> Process in which genetic information of an organism is changed, leading to mutation
            <br/>
            <u>Nuclease:</u> An enzyme that cleaves the chains of nucleotides in nucleic acids into smaller units.
            <br/>
            <u>Transcription: </u>A process in which DNA is read by an RNA polymerase and produces a complementary,
            antiparallel RNA strand called primary transcript.
            <br/>
            <u>Cleave: </u>A chemical reaction that breaks the DNA sequence into two at a specific point.
            <br/>
            <u>Phage: </u>A virus that parasitizes a bacterium by infecting it and reproducing inside it.
            <br/>
            <u>Homologous: </u>DNA that is identical or compatible.
            <br/>
            <u>Recombination: </u>An exchange between homologous region of DNA.
            <br/>
            <u>Oligonucleotide: </u>Polynucleotide whose molecules contain a relatively small number of nucleotides.
            <br/>
            <u>Point Mutation: </u>Mutation of a single nucleotide in the sequence.
            <br/>
            <u>Locus: </u>A portion of the genome in bacteria.
            <br/>
            <u>Protospacer: </u>Target sequence in the foreign genetic material.
            <br/>
            <u>Motif: </u>a sequence motif is a nucleotide or amino-acid sequence pattern that is widespread and has,
            or is
            thought to have, a biological significance.
            <br/>
            <u>PAM: </u>Precursor adjacent motif, part of CRISPR target sequence.
            <br/>
            <u>Endonuclease: </u>A nuclease that is present in the nucleus.
            <br/>
            <u>Non-homologous end joining:</u> after a dna strand is sliced, if there is no donor dna (donor dna:
            specific
            desired target dna) to insert in the space, the genes will repair by themselves which might cause a gene
            mutation due to a missing or additional base addition in the sequence. This type of repair is called
            non-homologous end joining.
            <br/>
            <u>Homology directed repair: </u>after a dna strand is sliced, if there is a homologous donor DNA present
            which is
            used to rejoin the sliced dna, it is called a homology directed repair.
            <br/>
            <u>Off-target: </u>refers to nonspecific and unintended genetic modifications
            <br/>
            <u>SgRNA: </u>Single guide RNA, it contains of two parts, a variable crRNA sequence which is usually 20
            base
            pairs long (read from 3&#8217; to 5&#8217; end and a constant tracrRNA that is used to bind to the Cas9
            enzyme which act as molecular scissors used to slice up double stranded DNA.
            <br/>
            <u>Heteroduplex: </u> heteroduplex is the more stable alignments with low percentage of mismatched base
            pairs.
            <br/>
            <u>Purines: </u>the bases adenine and guanine present in DNA and RNA.
          </p>
        </div>
      </div>
    </div>
  </section>

  <!--
    <section id="contact">
      <div class="container">
        <h2 class="text-center text-uppercase text-secondary mb-0">Contact Me</h2>
        <hr class="star-dark mb-5">
        <div class="row">
          <div class="col-lg-8 mx-auto">
            <form name="sentMessage" id="contactForm" novalidate="novalidate">
              <div class="control-group">
                <div class="form-group floating-label-form-group controls mb-0 pb-2">
                  <label>Name</label>
                  <input class="form-control" id="name" type="text" placeholder="Name" required="required" data-validation-required-message="Please enter your name.">
                  <p class="help-block text-danger"></p>
                </div>
              </div>
              <div class="control-group">
                <div class="form-group floating-label-form-group controls mb-0 pb-2">
                  <label>Email Address</label>
                  <input class="form-control" id="email" type="email" placeholder="Email Address" required="required" data-validation-required-message="Please enter your email address.">
                  <p class="help-block text-danger"></p>
                </div>
              </div>
              <div class="control-group">
                <div class="form-group floating-label-form-group controls mb-0 pb-2">
                  <label>Phone Number</label>
                  <input class="form-control" id="phone" type="tel" placeholder="Phone Number" required="required" data-validation-required-message="Please enter your phone number.">
                  <p class="help-block text-danger"></p>
                </div>
              </div>
              <div class="control-group">
                <div class="form-group floating-label-form-group controls mb-0 pb-2">
                  <label>Message</label>
                  <textarea class="form-control" id="message" rows="5" placeholder="Message" required="required" data-validation-required-message="Please enter a message."></textarea>
                  <p class="help-block text-danger"></p>
                </div>
              </div>
              <br>
              <div id="success"></div>
              <div class="form-group">
                <button type="submit" class="btn btn-primary btn-xl" id="sendMessageButton">Send</button>
              </div>
            </form>
          </div>
        </div>
      </div>
    </section> -->
  <!--
    <footer class="footer text-center">
      <div class="container">
        <div class="row">
          <div class="col-md-4 mb-5 mb-lg-0">
            <h4 class="text-uppercase mb-4">Location</h4>
            <p class="lead mb-0">2215 John Daniel Drive
              <br>Clark, MO 65243</p>
          </div>
          <div class="col-md-4 mb-5 mb-lg-0">
            <h4 class="text-uppercase mb-4">Around the Web</h4>
            <ul class="list-inline mb-0">
              <li class="list-inline-item">
                <a class="btn btn-outline-light btn-social text-center rounded-circle" href="#">
                  <i class="fab fa-fw fa-facebook-f"></i>
                </a>
              </li>
              <li class="list-inline-item">
                <a class="btn btn-outline-light btn-social text-center rounded-circle" href="#">
                  <i class="fab fa-fw fa-google-plus-g"></i>
                </a>
              </li>
              <li class="list-inline-item">
                <a class="btn btn-outline-light btn-social text-center rounded-circle" href="#">
                  <i class="fab fa-fw fa-twitter"></i>
                </a>
              </li>
              <li class="list-inline-item">
                <a class="btn btn-outline-light btn-social text-center rounded-circle" href="#">
                  <i class="fab fa-fw fa-linkedin-in"></i>
                </a>
              </li>
              <li class="list-inline-item">
                <a class="btn btn-outline-light btn-social text-center rounded-circle" href="#">
                  <i class="fab fa-fw fa-dribbble"></i>
                </a>
              </li>
            </ul>
          </div>
          <div class="col-md-4">
            <h4 class="text-uppercase mb-4">About Freelancer</h4>
            <p class="lead mb-0">Freelance is a free to use, open source Bootstrap theme created by
              <a href="http://startbootstrap.com">Start Bootstrap</a>.</p>
          </div>
        </div>
      </div>
    </footer> -->

  <div class="copyright py-4 text-center text-white">
    <div class="container">
      <small>Copyright &copy; Amna Irfan, Pratik Kshirsagar & Shivani Kamath</small>
    </div>
  </div>

  <!-- Scroll to Top Button (Only visible on small and extra-small screen sizes) -->
  <div class="scroll-to-top d-lg-none position-fixed ">
    <a class="js-scroll-trigger d-block text-center text-white rounded" href="#page-top">
      <i class="fa fa-chevron-up"></i>
    </a>
  </div>

  <!-- Portfolio Modals -->

  <!-- Portfolio Modal 1 -->
  <div class="portfolio-modal mfp-hide" id="portfolio-modal-1">
    <div class="portfolio-modal-dialog bg-white">
      <a class="close-button d-none d-md-block portfolio-modal-dismiss" href="#">
        <i class="fa fa-3x fa-times"></i>
      </a>
      <div class="container text-center">
        <div class="row">
          <div class="col-lg-8 mx-auto">
            <h2 class="text-secondary text-uppercase mb-0">Project Name</h2>
            <hr class="star-dark mb-5">
            <img class="img-fluid mb-5" src="img/portfolio/cabin.png" alt="">
            <p class="mb-5">Lorem ipsum dolor sit amet, consectetur adipisicing elit. Mollitia neque assumenda ipsam
              nihil,
              molestias magnam, recusandae quos quis inventore quisquam velit asperiores, vitae? Reprehenderit soluta,
              eos
              quod consequuntur itaque. Nam.</p>
            <a class="btn btn-primary btn-lg rounded-pill portfolio-modal-dismiss" href="#">
              <i class="fa fa-close"></i>
              Close Project</a>
          </div>
        </div>
      </div>
    </div>
  </div>

  <!-- Portfolio Modal 2 -->
  <div class="portfolio-modal mfp-hide" id="portfolio-modal-2">
    <div class="portfolio-modal-dialog bg-white">
      <a class="close-button d-none d-md-block portfolio-modal-dismiss" href="#">
        <i class="fa fa-3x fa-times"></i>
      </a>
      <div class="container text-center">
        <div class="row">
          <div class="col-lg-8 mx-auto">
            <h2 class="text-secondary text-uppercase mb-0">Project Name</h2>
            <hr class="star-dark mb-5">
            <img class="img-fluid mb-5" src="img/portfolio/cake.png" alt="">
            <p class="mb-5">Lorem ipsum dolor sit amet, consectetur adipisicing elit. Mollitia neque assumenda ipsam
              nihil,
              molestias magnam, recusandae quos quis inventore quisquam velit asperiores, vitae? Reprehenderit soluta,
              eos
              quod consequuntur itaque. Nam.</p>
            <a class="btn btn-primary btn-lg rounded-pill portfolio-modal-dismiss" href="#">
              <i class="fa fa-close"></i>
              Close Project</a>
          </div>
        </div>
      </div>
    </div>
  </div>

  <!-- Portfolio Modal 3 -->
  <div class="portfolio-modal mfp-hide" id="portfolio-modal-3">
    <div class="portfolio-modal-dialog bg-white">
      <a class="close-button d-none d-md-block portfolio-modal-dismiss" href="#">
        <i class="fa fa-3x fa-times"></i>
      </a>
      <div class="container text-center">
        <div class="row">
          <div class="col-lg-8 mx-auto">
            <h2 class="text-secondary text-uppercase mb-0">Project Name</h2>
            <hr class="star-dark mb-5">
            <img class="img-fluid mb-5" src="img/portfolio/circus.png" alt="">
            <p class="mb-5">Lorem ipsum dolor sit amet, consectetur adipisicing elit. Mollitia neque assumenda ipsam
              nihil,
              molestias magnam, recusandae quos quis inventore quisquam velit asperiores, vitae? Reprehenderit soluta,
              eos
              quod consequuntur itaque. Nam.</p>
            <a class="btn btn-primary btn-lg rounded-pill portfolio-modal-dismiss" href="#">
              <i class="fa fa-close"></i>
              Close Project</a>
          </div>
        </div>
      </div>
    </div>
  </div>

  <!-- Portfolio Modal 4 -->
  <div class="portfolio-modal mfp-hide" id="portfolio-modal-4">
    <div class="portfolio-modal-dialog bg-white">
      <a class="close-button d-none d-md-block portfolio-modal-dismiss" href="#">
        <i class="fa fa-3x fa-times"></i>
      </a>
      <div class="container text-center">
        <div class="row">
          <div class="col-lg-8 mx-auto">
            <h2 class="text-secondary text-uppercase mb-0">Project Name</h2>
            <hr class="star-dark mb-5">
            <img class="img-fluid mb-5" src="img/portfolio/game.png" alt="">
            <p class="mb-5">Lorem ipsum dolor sit amet, consectetur adipisicing elit. Mollitia neque assumenda ipsam
              nihil,
              molestias magnam, recusandae quos quis inventore quisquam velit asperiores, vitae? Reprehenderit soluta,
              eos
              quod consequuntur itaque. Nam.</p>
            <a class="btn btn-primary btn-lg rounded-pill portfolio-modal-dismiss" href="#">
              <i class="fa fa-close"></i>
              Close Project</a>
          </div>
        </div>
      </div>
    </div>
  </div>

  <!-- Portfolio Modal 5 -->
  <div class="portfolio-modal mfp-hide" id="portfolio-modal-5">
    <div class="portfolio-modal-dialog bg-white">
      <a class="close-button d-none d-md-block portfolio-modal-dismiss" href="#">
        <i class="fa fa-3x fa-times"></i>
      </a>
      <div class="container text-center">
        <div class="row">
          <div class="col-lg-8 mx-auto">
            <h2 class="text-secondary text-uppercase mb-0">Project Name</h2>
            <hr class="star-dark mb-5">
            <img class="img-fluid mb-5" src="img/portfolio/safe.png" alt="">
            <p class="mb-5">Lorem ipsum dolor sit amet, consectetur adipisicing elit. Mollitia neque assumenda ipsam
              nihil,
              molestias magnam, recusandae quos quis inventore quisquam velit asperiores, vitae? Reprehenderit soluta,
              eos
              quod consequuntur itaque. Nam.</p>
            <a class="btn btn-primary btn-lg rounded-pill portfolio-modal-dismiss" href="#">
              <i class="fa fa-close"></i>
              Close Project</a>
          </div>
        </div>
      </div>
    </div>
  </div>

  <!-- Portfolio Modal 6 -->
  <div class="portfolio-modal mfp-hide" id="portfolio-modal-6">
    <div class="portfolio-modal-dialog bg-white">
      <a class="close-button d-none d-md-block portfolio-modal-dismiss" href="#">
        <i class="fa fa-3x fa-times"></i>
      </a>
      <div class="container text-center">
        <div class="row">
          <div class="col-lg-8 mx-auto">
            <h2 class="text-secondary text-uppercase mb-0">Project Name</h2>
            <hr class="star-dark mb-5">
            <img class="img-fluid mb-5" src="img/portfolio/submarine.png" alt="">
            <p class="mb-5">Lorem ipsum dolor sit amet, consectetur adipisicing elit. Mollitia neque assumenda ipsam
              nihil,
              molestias magnam, recusandae quos quis inventore quisquam velit asperiores, vitae? Reprehenderit soluta,
              eos
              quod consequuntur itaque. Nam.</p>
            <a class="btn btn-primary btn-lg rounded-pill portfolio-modal-dismiss" href="#">
              <i class="fa fa-close"></i>
              Close Project</a>
          </div>
        </div>
      </div>
    </div>
  </div>

  <!-- Bootstrap core JavaScript -->
  <script src="vendor/jquery/jquery.min.js"></script>
  <script src="vendor/bootstrap/js/bootstrap.bundle.min.js"></script>

  <!-- Plugin JavaScript -->
  <script src="vendor/jquery-easing/jquery.easing.min.js"></script>
  <script src="vendor/magnific-popup/jquery.magnific-popup.min.js"></script>

  <!-- Contact Form JavaScript -->
  <script src="js/jqBootstrapValidation.js"></script>
  <script src="js/contact_me.js"></script>

  <!-- Custom scripts for this template -->
  <script src="js/freelancer.min.js"></script>

</body>

</html>